Feature/blueprint search vs extension sub tasks/create release vsix file (#8)
CoconutLizard/BlueprintSearchVSExtension: README.md

# BlueprintSearchVSExtension

A visual studio extension and associated UE plugin code to do a blueprint search without having to load up the editor.

Tested on engine versions: 4.26

##### Contents

* [Installing BlueprintSearchVS](#installing-blueprintsearchvs)

* [Instruction for Using BlueprintSearchVS](#using-blueprintsearchvs)

* [Instruction for Using Blueprint Search Filters](#using-blueprint-search-filters)

## Installing BlueprintSearchVS

> 1. Clone the repository inside your Unreal Engine plugins folder using your favourite Git client or by clicking the download ZIP button inside the Green Code submenu.

Reviewer: ```suggestion
1. Clone the repository inside your Unreal Engine plugins (<project path>/Engine/Plugins) folder using your favourite Git client or by clicking the download ZIP button inside the Green Code submenu.
```

2. Run GenerateProjectFiles from your Unreal Engine root folder and Rebuild your engine.

3. Close any open instance of Visual Studio.

> 4. Open BlueprintSearch.vsix

Reviewer: ```suggestion
4. Double click BlueprintSearch.vsix
```

Is this correct?

Reviewer 2: @Josef-CL  - Yes this is correct, any method of initializing the archive to run its installer will work (right clicking and selecting "open" will work too.

Furthermore, we could also add a step to select the relevant products for installing the extension (example installer window below). 
![VSIX_INSTALLER](https://user-images.githubusercontent.com/63600713/129875994-e3fa42f4-38d0-4a05-aded-37811e70959d.png)

5. Open Visual Studio and Rebuild your UE4 project.

> NOTE: If BlueprintSearchVS fails find one of the file needed to execute a search. Is possible, after fixing the error, to execute another search using the button inside the BlueprintSearchVS submenu.

Reviewer: I'm not completely sure what this means. Is it something the extension itself can provide instructions for if the problem happens (if it's not avoidable somehow)?

I mean the error box would say. Do X, Y, Z and then try again.

Author: Currently in these message boxes I'm only printing Error + How to solve it, not what to do afterwards. These error can be caused by users, i.e. in the case they forgot to build either engine or game project. Currently I'm searching for paths only when the extension is first initialized so if one of these errors occurs the user would have to restart visual studio, to solve this problem I added a button to search for Paths again. Not sure If I'm being clear if not I can try to explain myself better during the Catch-up tomorrow.

Reviewer 2: What you have commented makes sense @Davide-CL, I think it would be good to implement this note within the message box itself (as Josef stated), rather than in this document (or possibly include it in both), the note also needs re-articulating.

## Using BlueprintSearchVS

> 1. Open an Unreal Engine 4 project built against a version of the engine with the plugin in.

Reviewer: ```suggestion
1. Open an Unreal Engine 4 project built against a version of the engine with the BlueprintSearch plugin installed.
```

2. Open the Search Window from the Extensions drop down menu or by clicking Alt + Shift + Control + G.

3. Type the search terms inside the text box.

> 4. Click the search button or press Enter.

Reviewer 2: Also mention the keybinding shortcut here
